components/IngredientCard/components/editButton.tsx

```diff
@@ -0,0 +1,25 @@
+import Button from "@mui/joy/Button";
+import * as React from "react";
+
+interface EditButtonProps {
+    title: string;
+
+    onClick(): void;
+
+    type: "edit" | 'delete'
+}
+
+export const EditButton = ({title, onClick, type}: EditButtonProps) => {
```

Comment:
Sounds slightly over engineered. You can either put that definition into the `EditButtons` file, or use objects for shared props.



```jsx
const sharedProps = {
  variant: 'solid',
  size:'md'
}
// ...
export const EditButtons = ({onEdit, onDelete}: EditButtonsProps) => (
  <EditButton
    {...sharedProps}
    title={'Delete'}
    onClick={onDelete}
    type={'delete'}
  />
  <EditButton 
    {...sharedProps}
    title={"Edit"}
    onClick={onEdit}
    type={'edit'}
  />
)
```